be/src/storage/lake/compaction_task.cpp

```diff
@@ -42,4 +44,33 @@ Status CompactionTask::execute_index_major_compaction(TxnLogPB* txn_log) {
     return Status::OK();
 }
 
+void CompactionTask::fill_compaction_segment_info(TxnLogPB_OpCompaction* op_compaction, TabletWriter* writer) {
+    for (auto& rowset : _input_rowsets) {
+        op_compaction->add_input_rowsets(rowset->id());
+    }
+
+    // check last rowset whether this is a partial compaction
+    if (_tablet_schema->keys_type() != KeysType::PRIMARY_KEYS && _input_rowsets.size() > 0 &&
+        _input_rowsets.back()->partial_segments_compaction()) {
+        // add uncompacted files first, so that they will be compacted first next time
```

Comment:
uncompacted segments must be behind the compacted segments, because we should ensure the order of data, and this is necessary in unique key model.

Reply:
done